docs/architecture/maui/accessing-remote-data.md

```diff
@@ -1,9 +1,380 @@
+
+The `CatalogService` class is used to manage the data retrieval process from the catalog microservice. In the `RegisterViewModels` method in the `MauiProgram` class, the `CatalogService` class is registered as a type mapping against the `ICatalogService` type with the dependency injection container. Then, when an instance of the `CatalogViewModel` class is created, its constructor accepts an `ICatalogService type`, which the dependency injection container resolves, returning an instance of the `CatalogService` class. For more information about dependency injection, see [Dependency Injection](dependency-injection.md).
+
+The image below shows the interaction of classes that read catalog data from the catalog microservice for displaying by the CatalogView.
+
+:::image type="content" source="media/retrieving-data-for-catalog.png" alt-text="Retrieving data from the catalog microservice.":::
+
+When the `CatalogView` is navigated to, the `OnInitialize` method in the CatalogViewModel class is called. This method retrieves catalog data from the catalog microservice, as demonstrated in the following code example:
+
+```csharp
+public override async Task InitializeAsync(object navigationData)
+{
+    Products = await _productsService.GetCatalogAsync();
+} 
+```
+
+This method calls the `GetCatalogAsync` method of the `CatalogService` instance that was injected into the `CatalogViewModel` by the dependency injection container. The following code example shows the `GetCatalogAsync` method:
+
+```csharp
+public async Task<ObservableCollection<CatalogItem>> GetCatalogAsync()
+{
+    UriBuilder builder = new UriBuilder(GlobalSetting.Instance.CatalogEndpoint);
+    builder.Path = "api/v1/catalog/items";
+    string uri = builder.ToString();
+
+    CatalogRoot catalog = await _requestProvider.GetAsync<CatalogRoot>(uri);
+
+    return catalog?.Data;          
+} 
+```
+
+This method builds the URI that identifies the resource the request will be sent to, and uses the `RequestProvider` class to invoke the GET HTTP method on the resource, before returning the results to the `CatalogViewModel`. The `RequestProvider` class contains functionality that submits a request in the form of a URI that identifies a resource, an HTTP method that indicates the operation to be performed on that resource, and a body that contains any data required to perform the operation. For information about how the `RequestProvider` class is injected into the `CatalogService` class, see [Dependency Injection](dependency-injection.md).
+
+The following code example shows the `GetAsync` method in the `RequestProvider` class:
+
+```csharp
+public async Task<TResult> GetAsync<TResult>(string uri, string token = "")
+{
+    HttpClient httpClient = CreateHttpClient(token);
+    HttpResponseMessage response = await httpClient.GetAsync(uri);
+
+    await HandleResponse(response);
+    string serialized = await response.Content.ReadAsStringAsync();
+
+    TResult result = await Task.Run(() => 
```

Comment:
Do not use `Task.Run` when it is not needed. I'd argue that since the async state machine in this case has already yielded on the call to `.ReadAsStringAsync` that this should not be wrapped.